### ISSUE TITLE
mof_compiler flavors another issue

### ISSUE THREAD

**Issue by KSchopmeyer** [pywbem/pywbem#449]
the build_flavors function builds the flavor dictionary for a qualifier based on the input flavors list and the qualifier declaration flavors.  However, this is not correct.  The effective flavors for a qualifier within a class can only correctly be build based on the class hiearchy and the definition of the qualifier next highest in the hiearchy.  This function can set certain limits: disable override in the qualifier declaration cannot be converted to enable override in a qualifier for example.

However we should really be passing just the flavors defined in the input list and not trying to resolve themj completely.  As an example, the declaration could have set overridable but that could have been modified to non overridable further down in the class hiearchy. Without resolving this through the hiearchy the compiler cannot really resolve this.  

Since today we are using the compiler to build classes just for passing to external repositories, the code probably does not cause any problems but we should change it if we build a repository.

**Comment by andy-maier** [comment]
The flavors on qualifier (values) are for implicitly defined qualifier declarations. DSP0004 states that the scope for such implicitly defined qualifier declarations is the schema element that defines it. So the build_favors() function which apparently considers the flavor on the global qualifier declaration, and on the qualifier value in the defining schema element seems to be correct.

Unless someone thinks that the implicitly defined qualifier declarations would propagate down the hirrarchy. in which case this first would need to be discussed in context of DSP0004, and probably a spec clarification is needed.

Bottom line: We claim that the current implementation is correct and we close this issue.
